beacon-chain/state/stategen/migrate.go

```diff
@@ -55,24 +55,20 @@ func (s *State) MigrateToCold(ctx context.Context, fRoot [32]byte) error {
 				aRoot = cached.root
 				aState = cached.state
 			} else {
-				blks, err := s.beaconDB.HighestSlotBlocksBelow(ctx, slot)
+				_, roots, err := s.beaconDB.HighestRootsBelowSlot(ctx, slot)
 				if err != nil {
 					return err
 				}
 				// Given the block has been finalized, the db should not have more than one block in a given slot.
 				// We should error out when this happens.
-				if len(blks) != 1 {
+				if len(roots) != 1 {
```

Comment:
Unrelated to your PR but this check is wrong too... It appears that there is a lot of incorrect usage of `HighestRootsBelowSlot` across prysm. This is probably a source of many hidden bugs, we can fix this in a follow up PR.